tests/components/light/test_init.py

```diff
@@ -2,495 +2,623 @@
-
-        _, data = ent1.last_call("turn_on")
-        assert {} == data
-
-        _, data = ent2.last_call("turn_on")
-        assert {} == data
-
-        _, data = ent3.last_call("turn_on")
-        assert {} == data
-
-        # faulty attributes will not trigger a service call
-        common.turn_on(
-            self.hass, ent1.entity_id, profile=prof_name, brightness="bright"
+    _, data = ent3.last_call("turn_on")
+    assert data == {}
+
+    # faulty attributes will not trigger a service call
+    with pytest.raises(vol.MultipleInvalid):
+        await hass.services.async_call(
+            light.DOMAIN,
+            SERVICE_TURN_ON,
+            {
+                ATTR_ENTITY_ID: ent1.entity_id,
+                light.ATTR_PROFILE: prof_name,
+                light.ATTR_BRIGHTNESS: "bright",
+            },
+            blocking=True,
+        )
+    with pytest.raises(vol.MultipleInvalid):
+        await hass.services.async_call(
+            light.DOMAIN,
+            SERVICE_TURN_ON,
+            {
+                ATTR_ENTITY_ID: ent1.entity_id,
+                light.ATTR_RGB_COLOR: "yellowish",
+            },
+            blocking=True,
+        )
+    with pytest.raises(vol.MultipleInvalid):
+        await hass.services.async_call(
+            light.DOMAIN,
+            SERVICE_TURN_ON,
+            {ATTR_ENTITY_ID: ent2.entity_id, light.ATTR_WHITE_VALUE: "high"},
+            blocking=True,
         )
-        common.turn_on(self.hass, ent1.entity_id, rgb_color="yellowish")
-        common.turn_on(self.hass, ent2.entity_id, white_value="high")
 
-        self.hass.block_till_done()
+    _, data = ent1.last_call("turn_on")
+    assert data == {}
 
-        _, data = ent1.last_call("turn_on")
-        assert {} == data
+    _, data = ent2.last_call("turn_on")
+    assert data == {}
 
-        _, data = ent2.last_call("turn_on")
-        assert {} == data
 
-    def test_broken_light_profiles(self):
-        """Test light profiles."""
-        platform = getattr(self.hass.components, "test.light")
-        platform.init()
+async def test_broken_light_profiles(hass, mock_storage):
+    """Test light profiles."""
+    platform = getattr(hass.components, "test.light")
+    platform.init()
 
-        user_light_file = self.hass.config.path(light.LIGHT_PROFILES_FILE)
+    user_light_file = hass.config.path(light.LIGHT_PROFILES_FILE)
 
-        # Setup a wrong light file
-        with open(user_light_file, "w") as user_file:
-            user_file.write("id,x,y,brightness,transition\n")
-            user_file.write("I,WILL,NOT,WORK,EVER\n")
+    # Setup a wrong light file
+    with open(user_light_file, "w") as user_file:
+        user_file.write("id,x,y,brightness,transition\n")
+        user_file.write("I,WILL,NOT,WORK,EVER\n")
 
-        assert not setup_component(
-            self.hass, light.DOMAIN, {light.DOMAIN: {CONF_PLATFORM: "test"}}
-        )
+    assert not await async_setup_component(
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6239266)